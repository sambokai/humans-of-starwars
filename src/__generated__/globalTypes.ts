/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

/**
 *  The gender of this person. Will be "UNKNOWN" if not known or null if the person does not have a gender.
 */
export enum PERSON_GENDER {
  FEMALE = "FEMALE",
  HERMAPHRODITE = "HERMAPHRODITE",
  MALE = "MALE",
  UNKNOWN = "UNKNOWN",
}

//==============================================================
// END Enums and Input Objects
//==============================================================
