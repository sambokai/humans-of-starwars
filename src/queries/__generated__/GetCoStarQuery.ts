/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: GetCoStarQuery
// ====================================================

export interface GetCoStarQuery_allPersons_films {
  __typename: "Film";
  id: string;
  /**
   * The episode number of this film.
   */
  episodeId: number;
}

export interface GetCoStarQuery_allPersons {
  __typename: "Person";
  id: string;
  /**
   * The name of this person.
   */
  name: string;
  films: GetCoStarQuery_allPersons_films[] | null;
}

export interface GetCoStarQuery_Person_films {
  __typename: "Film";
  id: string;
  /**
   * The episode number of this film.
   */
  episodeId: number;
}

export interface GetCoStarQuery_Person {
  __typename: "Person";
  id: string;
  films: GetCoStarQuery_Person_films[] | null;
}

export interface GetCoStarQuery {
  allPersons: GetCoStarQuery_allPersons[];
  Person: GetCoStarQuery_Person | null;
}

export interface GetCoStarQueryVariables {
  id: string;
}
