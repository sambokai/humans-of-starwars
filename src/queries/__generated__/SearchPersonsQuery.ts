/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: SearchPersonsQuery
// ====================================================

export interface SearchPersonsQuery_allPersons {
  __typename: "Person";
  id: string;
  /**
   * The name of this person.
   */
  name: string;
}

export interface SearchPersonsQuery {
  allPersons: SearchPersonsQuery_allPersons[];
}

export interface SearchPersonsQueryVariables {
  queryString: string;
}
