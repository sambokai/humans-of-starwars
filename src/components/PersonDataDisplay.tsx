import React from 'react';
import {PERSON_GENDER} from '../__generated__/globalTypes';
import getGenderString from "../queries/utils/getGenderString";

interface Props {
    id: string;
    name: string;
    gender?: PERSON_GENDER;
    homeworldName?: string;
    appearedInFilms?: Film[];
}

interface Film {
    title: string;
    releaseDate: Date;
}



const PersonDataDisplay: React.FunctionComponent<Props> = ({name, gender, homeworldName, appearedInFilms, children}) => {


    return (
        <article>
            <h2>{name}</h2>
            <h4>{getGenderString(gender)} {homeworldName && `from ${homeworldName}`}</h4>


            {appearedInFilms && <ul>
                <p>Appeared in:</p>
                {appearedInFilms.map((film, i) => <li key={i}>{film.title} ({film.releaseDate.getFullYear()})</li>)}
            </ul>}

            {children}

        </article>
    )
};

export default PersonDataDisplay;
