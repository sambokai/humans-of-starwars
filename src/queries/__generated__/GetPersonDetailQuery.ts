/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.


// ====================================================
// GraphQL query operation: GetPersonDetailQuery
// ====================================================

import {PERSON_GENDER} from "../../__generated__/globalTypes";

export interface GetPersonDetailQuery_Person_species {
  __typename: "Species";
  /**
   * The name of this species.
   */
  name: string;
}

export interface GetPersonDetailQuery_Person_homeworld {
  __typename: "Planet";
  /**
   * The name of the planet
   */
  name: string;
}

export interface GetPersonDetailQuery_Person_films {
  __typename: "Film";
  /**
   * The title of this film
   */
  title: string;
  /**
   *  The ISO 8601 date format of film release at original creator country.
   */
  releaseDate: any | null;
}

export interface GetPersonDetailQuery_Person {
  __typename: "Person";
  id: string;
  /**
   * The name of this person.
   */
  name: string;
  /**
   *  The gender of this person. Will be "UNKNOWN" if not known or null if the person does not have a gender.
   */
  gender: PERSON_GENDER | null;
  species: GetPersonDetailQuery_Person_species[] | null;
  homeworld: GetPersonDetailQuery_Person_homeworld | null;
  films: GetPersonDetailQuery_Person_films[] | null;
}

export interface GetPersonDetailQuery {
  Person: GetPersonDetailQuery_Person | null;
}

export interface GetPersonDetailQueryVariables {
  id: string;
}
