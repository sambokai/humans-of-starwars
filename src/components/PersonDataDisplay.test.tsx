import React from 'react';
import {shallow} from "enzyme";
import PersonDataDisplay from "./PersonDataDisplay";
import {PERSON_GENDER} from "../__generated__/globalTypes";

const mockPerson = {
    id: "1",
    name: "Luke Skywalker",
    homeWorld: "Tatooine",
    gender: PERSON_GENDER.MALE,
    films: [
        {title: "Movie A", releaseDate: new Date(2000, 1)},
        {title: "Movie B", releaseDate: new Date(2000, 2)}
    ]
};

describe('PersonDataDisplay component', () => {
    it('should render correctly', () => {
        const wrapper = shallow((
            <PersonDataDisplay id={mockPerson.id} name={mockPerson.name}/>
        ));

        expect(wrapper).toMatchSnapshot();
    });

    describe('when given a homeworld name', () => {
        it('should render correctly', () => {
            const wrapper = shallow((
                <PersonDataDisplay id={mockPerson.id} name={mockPerson.name} homeworldName={mockPerson.homeWorld}/>
            ));

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when given a gender', () => {
        it('should render correctly', () => {
            const wrapper = shallow((
                <PersonDataDisplay id={mockPerson.id} name={mockPerson.name} gender={mockPerson.gender}/>
            ));

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when given both a homeworld and a gender', () => {
        it('should render correctly', () => {
            const wrapper = shallow((
                <PersonDataDisplay id={mockPerson.id} name={mockPerson.name} gender={mockPerson.gender} homeworldName={mockPerson.gender}/>
            ));

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when given a list of films', () => {
        it('should render correctly', () => {
            const wrapper = shallow((
                <PersonDataDisplay id={mockPerson.id} name={mockPerson.name} appearedInFilms={mockPerson.films}/>
            ));

            expect(wrapper).toMatchSnapshot();
        });
    });
});
