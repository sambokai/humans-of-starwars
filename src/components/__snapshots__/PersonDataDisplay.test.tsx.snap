// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDataDisplay component should render correctly 1`] = `
<article>
  <h2>
    Luke Skywalker
  </h2>
  <h4>
     
  </h4>
</article>
`;

exports[`PersonDataDisplay component when given a gender should render correctly 1`] = `
<article>
  <h2>
    Luke Skywalker
  </h2>
  <h4>
    Male
     
  </h4>
</article>
`;

exports[`PersonDataDisplay component when given a homeworld name should render correctly 1`] = `
<article>
  <h2>
    Luke Skywalker
  </h2>
  <h4>
     
    from Tatooine
  </h4>
</article>
`;

exports[`PersonDataDisplay component when given a list of films should render correctly 1`] = `
<article>
  <h2>
    Luke Skywalker
  </h2>
  <h4>
     
  </h4>
  <ul>
    <p>
      Appeared in:
    </p>
    <li
      key="0"
    >
      Movie A
       (
      2000
      )
    </li>
    <li
      key="1"
    >
      Movie B
       (
      2000
      )
    </li>
  </ul>
</article>
`;

exports[`PersonDataDisplay component when given both a homeworld and a gender should render correctly 1`] = `
<article>
  <h2>
    Luke Skywalker
  </h2>
  <h4>
    Male
     
    from MALE
  </h4>
</article>
`;
